app/presenters/layouts/app_router_presenter.rb

```diff
@@ -0,0 +1,228 @@
+module Layouts
+  class AppRouterPresenter < ::ApplicationPresenter
+    def initialize(view_context, course)
+      @course = course
+      super(view_context)
+    end
+
+    private
+
+    def props
+      {
+        courses: course_details_array,
+        current_user: user_details,
+        school: school_details
+      }
+    end
+
+    # private
+
+    def school_details
+      { name: school_name, logo_url: logo_url, links: nav_links }
+    end
+
+    def user_details
+      user = {
+        name: current_user.name,
+        title: current_user.full_title,
+        is_admin: current_school_admin.present?,
+        can_edit_profile: show_user_edit?,
+        has_notifications: notifications?
+      }
+      if current_user.avatar.attached?
+        user[:avatar_url] = view.url_for(current_user.avatar_variant(:thumb))
+      end
+
+      user[:coach_id] = current_coach.id if current_coach.present?
+      user
+    end
+
+    def courses
+      if current_user.blank?
+        current_school.courses.live.where(public_preview: true)
+      elsif current_school_admin.present?
+        # All courses are available to admins.
+        current_school.courses.live
+      else
+        # current course if course has public preview.
+        previewed_course = @course.public_preview? ? [@course] : []
+
+        (
+          courses_with_author_access + courses_with_review_access +
+            courses_with_student_profile + previewed_course
+        ).uniq
+      end
+    end
+
+    def courses_with_student_profile
+      @courses_with_student_profile ||=
+        begin
+          Course
+            .joins(:founders)
+            .where(
+              school: current_school,
+              founders: {
+                id: current_user.founders.select(:id)
+              }
+            )
+        end.to_a
```

Comment:
I think the `begin` and `end` keywords are unnecessary:

```suggestion
      @courses_with_student_profile ||=
        Course
          .joins(:founders)
          .where(
            school: current_school,
            founders: {
              id: current_user.founders.select(:id)
            }
          )
          .to_a
```